src/main/java/frc/robot/ChallengeFive.java

```diff
@@ -17,160 +17,161 @@
+            if(xController.getY(Hand.kLeft) != 0 || xController.getX(Hand.kLeft) != 0)
+            { // if the joystick is not idle
+                if(xController.getY(Hand.kLeft) > 0 && xController.getY(Hand.kLeft) > Math.abs(xController.getX(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(FULLPOWER, FULLPOWER); // move up
+                }
+                if(xController.getY(Hand.kLeft) < 0 && Math.abs(xController.getX(Hand.kLeft))< Math.abs(xController.getY(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(FULLPOWER_neg, FULLPOWER_neg);// move down
+                }
+                if(xController.getX(Hand.kLeft) < 0 && Math.abs(xController.getY(Hand.kLeft))< Math.abs(xController.getX(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(TURNPOWER, FULLPOWER);// move left
+                }
+                if(xController.getX(Hand.kLeft) > 0 && Math.abs(xController.getY(Hand.kLeft))< Math.abs(xController.getX(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(FULLPOWER, TURNPOWER); //move right
+                }
             }
-            if(xController.getY(Hand.kLeft) > 0 && Math.abs(xController.getX(Hand.kLeft))> Math.abs(xController.getY(Hand.kLeft)))
-            {
-                controller.setDriveSpeed(FULLPOWER_neg, FULLPOWER_neg);// move up
+            else{
+                controller.setDriveSpeed(ZERO, ZERO); // stop motors/idle
             }
-            if(xController.getX(Hand.kLeft) > 0 && Math.abs(xController.getY(Hand.kLeft)) > Math.abs(xController.getX(Hand.kLeft)))
-            {
-                controller.setDriveSpeed(TURNPOWER, FULLPOWER);// move right
+        }
+        while(inverse == true)
+        {  //These are inverse controls
+            if(xController.getY(Hand.kLeft) != 0 || xController.getX(Hand.kLeft) != 0)
+            { // if the joystick is not idle
+                if(xController.getY(Hand.kLeft) < 0 && xController.getY(Hand.kLeft) < Math.abs(xController.getX(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(FULLPOWER, FULLPOWER); // move down
+                }
+                if(xController.getY(Hand.kLeft) > 0 && Math.abs(xController.getX(Hand.kLeft))> Math.abs(xController.getY(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(FULLPOWER_neg, FULLPOWER_neg);// move up
+                }
+                if(xController.getX(Hand.kLeft) > 0 && Math.abs(xController.getY(Hand.kLeft)) > Math.abs(xController.getX(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(TURNPOWER, FULLPOWER);// move right
+                }
+                if(xController.getX(Hand.kLeft) < 0 && Math.abs(xController.getY(Hand.kLeft)) < Math.abs(xController.getX(Hand.kLeft)))
+                {
+                    controller.setDriveSpeed(FULLPOWER, TURNPOWER); //move left
+                }
```

Comment:
I don't think we want to only be able to go 4 directions... Is there reasoning behind this decision? Please write comments explaining the controls.